pkg/resource/restconfig_factory.go

```diff
@@ -10,6 +13,19 @@ import (
 )
 
 func (r *Helper) getRESTConfigFactory(namespace string) (cmdutil.Factory, error) {
+	if r.metricsEnabled {
+		// Copy the possibly shared restConfig reference and add a metrics wrapper.
+		cfg := rest.CopyConfig(r.restConfig)
+		cfg.WrapTransport = func(rt http.RoundTripper) http.RoundTripper {
```

Comment:
@dgoodwin I think you should take into consideration the case where WrapTransport is already set to some value. I don't think that happens in any of our use cases currently, but it does happen. I saw it when we were building a client from a kubeconfig pointing to service account credentials.

Reply:
Yeah I tested and it was unset but I can see if I can get it to wrap the existing.